src/Map/ProspectorOverlayLayer.cs

```diff
@@ -57,14 +57,12 @@ public ProspectorOverlayLayer(ICoreAPI api, IWorldMapManager mapSink) : base(api
                         invMan.SlotModified += Event_SlotModified;
                     }
                 };
-                _clientApi.Event.PlayerLeave += (p) =>
-                {
-                    if (p == _clientApi?.World.Player)
-                    {
-                        var invMan = p?.InventoryManager?.GetHotbarInventory();
-                        invMan.SlotModified -= Event_SlotModified;
-                    }
+
+                // Save data when leaving and periodically.
+                _clientApi.Event.LeaveWorld += () => {
+                    SaveProspectingData();
                 };
+                _clientApi.World.RegisterGameTickListener(SaveProspectingData, (int) TimeSpan.FromMinutes(10).TotalMilliseconds);
```

Comment:
i don't think we need to be that conservativ regarding calling this. I would prefer either configuration or a much lower value since it's just a flag check in most cases if the user is not prospecting and otherwise we want to save it a short time after (like a minute or so)